### DISCUSSION TITLE
How to serialize SDK Output struct?

### DISCUSSION THREAD

**Discussion by Samrose-Ahmed** [awslabs/aws-sdk-rust#714] [Q&A] [answered]
I am using the Inspector V2 SDK (https://crates.io/crates/aws-sdk-inspector2)

When I call `list_findings`, it returns a `ListFindingsOutput`.

But I want to send this to a different system, and now its parsed to into a Rust struct. 

How can I serialize it back to JSON (or get the raw json response)?

**Comment by rcoh** [comment, marked as answer]
Unfortunately, we don't currently support serde (although there is an ongoing community contribution to enable that: [RFC 30: serde support](https://awslabs.github.io/smithy-rs/design/rfcs/rfc0030_serialization_and_deserialization.html)).

It's possible to get access to the raw response, however, I'd caution you from going that route. The raw response may contain fields serialized in headers—just looking at the body won't necessarily contain everything you need.

I'd probably suggest making your own parallel struct that contains the fields you need. In this case, `ListFindingsOutput` only contains one meaningful field—`finding_arns` which is the list of `arns`.

Please note that this API is actually [paginated](https://docs.rs/aws-sdk-inspector/latest/aws_sdk_inspector/client/fluent_builders/struct.ListFindings.html#method.into_paginator) so `ListFindingsOutput` isn't what you'd want to serialize anyway. 


```rust
#[derive(Serialize, Deserialize)]
struct Findings { arns: Vec<String> }
async fn serialize_findings(client: &aws_sdk_inspector2::Client) -> Result<Findings, aws_sdk_inspector2::Error> {
  
  let mut arns: Vec<String> = vec![];
  let mut findings_stream = client.list_findings()/*.<params>*/.into_paginator().send();
  while let Some(findings) = findings_stream.try_next().await? {
     arns.extend(findings.findings().unwrap_or_default().iter().map(|finding|
        // just copy the ARN so it can be retrieved. More fields could also be copied.
        finding.finding_arn().unwrap_or_default().to_string())
    );
  }
  Ok( Findings { arns })
}

**Reply by Samrose-Ahmed** [reply]
Thanks I actually had to use  Inspector2 API which returns `findings` list of massive structs, I ended up using the raw response (https://github.com/matanolabs/matano/pull/61) which seems to work fine.
